src/main/java/org/cbioportal/legacy/web/PublicVirtualStudiesController.java

```diff
@@ -113,13 +153,28 @@ private void unPublishVirtualStudy(String id) {
           "The virtual study with id=" + id + " has not been found in the public list.");
     }
     VirtualStudyData virtualStudyData = virtualStudyToUnPublish.getData();
+    checkIfVSWasPublished(id, virtualStudyData);
+    virtualStudyData.setUsers(Set.of(virtualStudyData.getOwner()));
+    sessionServiceRequestHandler.updateVirtualStudy(virtualStudyToUnPublish);
+  }
+
+  /**
+   * Drops public virtual study, removing it from the public list
+   *
+   * @param id - id of public virtual study to drop
+   */
+  private void dropPublicVirtualStudy(String id) {
+    VirtualStudy virtualStudyToUnPublish = sessionServiceRequestHandler.getVirtualStudyById(id);
+    checkIfVSWasPublished(id, virtualStudyToUnPublish.getData());
+    sessionServiceRequestHandler.dropVirtualStudy(virtualStudyToUnPublish.getId());
```

Comment:
naming is a bit confusing here. So we have `getVirtualStudyById(id);` and `dropVirtualStudy(virtualStudyToUnPublish.getId())`. Why not `dropVirtualStudy(id)`?

Reply:
@pieterlukasse Done https://github.com/cBioPortal/cbioportal/pull/11611/commits/0a430cf8151c861cb06add43b02b17d7c102a9e8